docs/content/latest/api/ysql/datatypes/type_array/literals/text-typecasting-and-literals.md

```diff
@@ -0,0 +1,242 @@
+  end;
+
+  -- 1-d array of "row" type values
+  declare
+    original   constant rt[]   not null :=
+      array[
+        row(42.1763, 'dog house', now(),                    true),
+        row(19.8651, 'cat flap',  now() + interval '1' day, false)
+      ];
+    text_cast  constant text   not null := original::text;
+    recreated  constant rt[]   not null := text_cast::rt[];
+  begin
+    assert
+      (recreated = original),
+    'assert failed';
+    raise info 'array of "row" type:  %', text_cast;
+  end;
+end;
+$body$;
+```
+It produces this result (after manually removing the `INFO:` prompt on each output line.
+```
+numeric:              42.1763
+timestamp:            2020-05-03 22:25:42.932771
+timestamptz:          2020-05-03 22:25:42.932771-07
+boolean:              true
+"row" type:           (42.1763,"dog house","2020-05-03 22:25:42.932771",t)
+2-d array             {{1,2},{3,4}}
+array of "row" type:  {"(42.1763,\"dog house\",\"2020-05-03 22:25:42.932771\",t)","(19.8651,\"cat flap\",\"2020-05-04 22:25:42.932771\",f)"}
+```
+The remaining sections in the present _"[Creating an array value using a literal](../../literals/)"_ enclosing section explain the syntax of the last three `text` values.
+
+## boolean values show special text forms in _ysqlsh_
+
+Try this:
+```postgresql
+select true as "bare true", true::text as "true::text";
+```
+This is the result:
+```
+ bare true | true::text 
+-----------+------------
+ t         | true
+```
+For all but `boolean` values, the string of characters that _ysqlsh_ uses to display any value is the `::text` typecast of that value. (After all, the only feasible means of display is strings of characters.) But uniquely for the two `boolean` values `true` and `false` it uses the single characters `t` and `f` rather than their `::text` typecasts—unless you explicitly write the typecast. 
+
+This behavior is inherited from `psql`.
+
+We saw above that even when you explicitly `::text` typecast a composite value, `true` and `false` are represented as `t` and `f`. You can't influence this outcome because it has to do with the rules for deriving the `text` of the typecast and _not_ with the convention that _ysqlsh_ uses. This asymmetry was established very many years ago; and it will not change.
+
+## The relationship between the text typecast of a value and the literal that creates that value
+
+Try this in _ysqlsh_:
+```postgresql
+select
+  42.932771::numeric          as n,
+  'cat'::text                 as t1,
+  $$dog's breakfast$$::text   as t2,
+  array[1, 2, 3]::int[]       as "int array";
+```
+It shows the result:
+```
+     n     | t1  |       t2        | int array 
+-----------+-----+-----------------+-----------
+ 42.932771 | cat | dog's breakfast | {1,2,3}}
+```
+ You won't be surprised by this. But we need to establish the proper terms of art that allow us to describe what's going on precisely and correctly. The remaining sections in the present enclosing section rely on this.
```

Comment:
```suggestion
 You won't be surprised by this. But you need to establish the proper terms of art that allow you to describe what's going on precisely and correctly. The remaining sections in the present enclosing section rely on this.
```

Reply:
Oh dear. This seems wrong to me. It's me that has to establish stuff so that I can explain things properly. In the old days, I've written it all in the passive voice. "It's essential first to establish the proper... so that whats going on can be described..." I'm not instructing the user to select this, click that, and then establish the other!